Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs

```diff
@@ -762,6 +763,100 @@ public IEnumerator TestInteractableToggleCollectionIndexUpdate()
             GameObject.Destroy(interactableToggleCollection.gameObject);
         }
 
+
+        /// <summary>
+        /// Test if Button state is reset when it goes out of focus from a pressed state
+        /// </summary>
+        [UnityTest]
+        public IEnumerator TestButtonStateResetWhenFocusLostAfterPinch()
+        {
+
+            Interactable interactable;
+            Transform interactableTransform;
+
+            InstantiatePressableButtonHoloLens2(
+                new Vector3(0.0f, 0.1f, 0.4f),
+                DefaultRotationToggle,
+                out interactable,
+                out interactableTransform);
+
+            Assert.True(interactable.IsEnabled);
+
+            var rightHand = new TestHand(Handedness.Right);
+            Vector3 focusPosition = new Vector3(0.015f, 0.015f, 0.3f);
+            Vector3 releaseDelta = new Vector3(0.05f, 0, 0);
+
+
+            // Initial Button State On Focus from Far Interaction Pointer
+
+            yield return rightHand.Show(focusPosition);
+            yield return PlayModeTestUtilities.WaitForInputSystemUpdate();
+
+            Assert.True(interactable.HasFocus);
+            Assert.False(interactable.HasPress);
+            Assert.False(interactable.HasGesture);
+            Assert.True(interactable.StateManager.CurrentState().Name == "Focus", "Interactable State is not Focus");
+
+
+            // Pinch Gesture and keeping focus
+
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            yield return PlayModeTestUtilities.WaitForInputSystemUpdate();
+
+            Assert.True(interactable.HasFocus);
+            Assert.True(interactable.HasPress);
+            Assert.False(interactable.HasGesture);
+            Assert.True(interactable.StateManager.CurrentState().Name == "Pressed", "Interactable State is not Pressed");
+
+
+            // Move Hand to make button out of focus. Button State should go to Default State
+
+            yield return rightHand.Move(releaseDelta);
+            yield return PlayModeTestUtilities.WaitForInputSystemUpdate();
```

Comment:
I noticed that the press-off code has some time delta based waiting - do you know if there's any possible flakiness here due to this?

i.e. is it possible that a single update isn't actually enough for the press off/gesture off handling code to run (because it's waiting for time delta?)

Reply:
I believe this is being tested on the assert that follows that WaitForInputSystemUpdate(), no?

Reply:
Yessss but also maybe. The only reason I bring this up is because this rolloff mechanism is based on time delta - it looks like the 10 frames that get waited in WaitForInputSystemUpdate() is sufficient, I'd just be curious to see going forward if we see flakiness in the test.

So based on this, no action to take now, we can just watch out for test flakiness in CI going forward (and there might be none)